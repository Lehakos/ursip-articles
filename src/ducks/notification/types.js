export const SHOW_NOTIFICATION = 'app/SHOW_NOTIFICATION';
export const HIDE_NOTIFICATION = 'app/HIDE_NOTIFICATION';
export const CHANGE_NOTIFICATION = 'app/CHANGE_NOTIFICATION';
