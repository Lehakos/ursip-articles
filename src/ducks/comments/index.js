import reducer from './reducers';

export * as constants from './constants';
export * as actions from './actions';
export * as selectors from './selectors';
export { default as saga } from './sagas';

export default reducer;
