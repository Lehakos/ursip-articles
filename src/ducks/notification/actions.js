import * as types from './types';

export const showNotification = (data) => ({
  type: types.SHOW_NOTIFICATION,
  payload: { data },
});

export const hideNotification = () => ({
  type: types.HIDE_NOTIFICATION,
});

export const changeNotification = text => ({
  type: types.CHANGE_NOTIFICATION,
  payload: { text },
});
