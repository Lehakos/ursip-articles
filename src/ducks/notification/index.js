import reducer from './reducers';

export * as actions from './actions';
export * as selectors from './selectors';

export default reducer;
