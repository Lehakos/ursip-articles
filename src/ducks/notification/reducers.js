import { fromJS } from 'immutable';

import * as types from './types';

const initialState = fromJS({
  show: false,
  text: '',
});

export default function (state = initialState, action) {
  const { type, payload } = action;

  switch (type) {
    case types.CHANGE_NOTIFICATION:
      return state.set('text', payload.text);

    case types.SHOW_NOTIFICATION:
      return state
        .set('text', payload.text)
        .set('show', true);

    case types.HIDE_NOTIFICATION:
      return state
        .set('text', '')
        .set('show', false);

    default:
      return state;
  }
}
