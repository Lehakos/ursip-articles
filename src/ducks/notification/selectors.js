import { createSelector } from 'reselect';

const getDomainState = state => state.get('notification');

const makeSelectNotificationState = () => createSelector(
  getDomainState,
  notif => notif.toJS(),
);

export {
  makeSelectNotificationState,
};
