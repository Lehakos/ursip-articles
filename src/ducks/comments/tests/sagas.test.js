import { put } from 'redux-saga/effects';

import { actions as notificationActions } from 'ducks/notification';

import {
  getComments as getCommentsAction,
  getCommentsSuccess,
  getCommentsFail,
} from '../actions';
import {
  getComments,
  GET_COMMENTS_ERROR,
} from '../sagas';
import commentsData, { normalized } from './mocks';

describe('getComments сага', () => {
  let getCommentsGen;

  beforeEach(() => {
    getCommentsGen = getComments(getCommentsAction(1));
    const callDescriptor = getCommentsGen.next().value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('если комментарии успешно загружены, должна вызываться getCommentsSuccess', () => {
    const response = commentsData;

    const putDescriptor = getCommentsGen.next({ data: response }).value;
    expect(putDescriptor).toEqual(put(getCommentsSuccess(normalized)));
  });

  it('если произошла ошибка, должно быть показано уведомление и затем вызвана getCommentsFail', () => {
    const notificationDescriptor = getCommentsGen.throw('Ошибка').value;
    expect(notificationDescriptor).toEqual(
      put(notificationActions.showNotification({ text: GET_COMMENTS_ERROR })),
    );

    const putDescriptor = getCommentsGen.next().value;
    expect(putDescriptor).toEqual(put(getCommentsFail()));
  });
});
