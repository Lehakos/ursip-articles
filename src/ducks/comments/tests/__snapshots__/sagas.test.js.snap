// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getComments сага если комментарии успешно загружены, должна вызываться getCommentsSuccess 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      1,
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`getComments сага если произошла ошибка, должно быть показано уведомление и затем вызвана getCommentsFail 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      1,
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
